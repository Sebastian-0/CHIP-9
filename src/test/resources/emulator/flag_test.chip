
clrflag
sout FLAGS

# Set all flags
setflag Z, 1
sout FLAGS

setflag N, 1
sout FLAGS

setflag H, 1
sout FLAGS

setflag C, 1
sout FLAGS

# Clear all flags
setflag Z, 0
sout FLAGS

setflag N, 0
sout FLAGS

setflag H, 0
sout FLAGS

setflag C, 0
sout FLAGS

# Test clrflags
setflag Z, 1
setflag N, 1
setflag H, 1
setflag C, 1

clrflag
sout FLAGS

hcf