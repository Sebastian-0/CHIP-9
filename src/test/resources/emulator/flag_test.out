0

128
192
224
240

112
48
16
0

0